frame/ethereum/src/lib.rs

```diff
@@ -187,12 +188,51 @@ decl_module! {
 
 			let block = ethereum::Block {
 				header,
-				transactions,
+				transactions: transactions.clone(),
 				ommers,
 			};
 
 			BlocksAndReceipts::insert(hash, (block, receipts));
 			BlockNumbers::<T>::insert(n, hash);
+
+			for t in &transactions {
+				let transaction_hash = H256::from_slice(
+					Keccak256::digest(&rlp::encode(t)).as_slice()
+				);
+				if let Some(transaction_status) = TransactionStatuses::get(transaction_hash) {
```

Comment:
Is it possible that we store the mapping as just `transaction_hash => (block_hash, transaction_index)`? That would make future transition into proper store-ethblock-in-substrate-digests much easier